File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.java
import java.util.*;

class Solution {
    public int[] solution(int[] arr) {
        int[] answer;
        if(arr.length > 1){
            answer = new int[arr.length - 1];
            int min = arr[0];
            for(int i = 0; i < arr.length; i++){
                if(arr[i] < min){
                    min = arr[i];
                }
            }
            int count = 0;
            for(int i = 0; i < arr.length; i++){
                if(arr[i] != min){
                    answer[count] = arr[i];
                    count++;
                }
            }
        }else{
            answer = new int[1];
            answer[0] = -1;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/12935. 제일 작은 수 제거하기/제일 작은 수 제거하기.swift
func solution(_ arr:[Int]) -> [Int] {
    var result = arr
    
    if arr.count == 1 {
        result.removeAll()
        result.append(-1)
        return result
    }
    
    var num = result[0]
    
    for i in result {
        if num > i {
            num = i
        }
    }
    
    if let value = result.firstIndex(of: num) {
        result.remove(at: value)
    }
    
    return result
}